extensions/sql-bindings/src/services/azureFunctionsService.ts

```diff
@@ -71,41 +70,38 @@ export async function createAzureFunction(connectionString: string, schema: stri
 				return;
 			}
 
-			// create C# HttpTrigger
+			// select input or output binding
+			const selectedBinding = await azureFunctionsUtils.promptForBindingType();
+
+			if (!selectedBinding) {
+				return;
+			}
+
+			// set the templateId based on the selected binding type
+			let templateId: string = selectedBinding.type === BindingType.input ? constants.inputTemplateID : constants.outputTemplateID;
+			let objectName = utils.generateQuotedFullName(schema, table);
+
+			// create C# Azure Function with SQL Binding
 			await azureFunctionApi.createFunction({
 				language: 'C#',
-				templateId: 'HttpTrigger',
+				templateId: templateId,
 				functionName: functionName,
+				functionSettings: {
+					connectionStringSetting: constants.sqlConnectionStringSetting,
+					...(selectedBinding.type === BindingType.input && { object: objectName }),
+					...(selectedBinding.type === BindingType.output && { table: objectName })
+				},
 				folderPath: projectFile
 			});
 
 			// check for the new function file to be created and dispose of the file system watcher
 			const timeoutForFunctionFile = utils.timeoutPromise(constants.timeoutAzureFunctionFileError);
-			functionFile = await Promise.race([newFunctionFileObject.filePromise, timeoutForFunctionFile]);
+			await Promise.race([newFunctionFileObject.filePromise, timeoutForFunctionFile]);
 		} finally {
 			newFunctionFileObject.watcherDisposable.dispose();
 		}
-
-		// select input or output binding
-		const selectedBinding = await azureFunctionsUtils.promptForBindingType();
-
-		if (!selectedBinding) {
-			return;
-		}
-
 		await azureFunctionsUtils.addNugetReferenceToProjectFile(projectFile);
```

Comment:
Do we still need this? I thought the template added this for us as well.

Reply:
Yes, just checked this. It does add the NuGet reference.

`<PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Sql" Version="0.1.131-preview" />`